elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/JoinTo.java

```diff
@@ -24,5 +25,13 @@
      * If the current entity is author, then a path would be "book.publisher.name".
      * @return The path
      */
-    String path();
+    String path() default "";
+
+    /**
+     * Join on clause constraints for customizing relationship joins.
+     * Use "%from" and "%join% to represent the two sides of join.
+     *
+     * @return join constraint like <code>%from.col1 = %join.col2</code>
+     */
+    JoinExpression[] constraints() default {};
```

Comment:
The join clause could use OR instead of AND to join them.  Since we break out constraints rather than just the ON clause - we have to pick (likely AND) and ignore OR.  I think it would be better to just have a single ON clause expression (rather than breaking this up).  Another thing that is missing is parenthetical expressions.  The more structure we add to the ON clause - the more complicated the configuration.  We should just let them supply the ON clause.